testnet/stacks-node/src/tests/mod.rs

```diff
@@ -341,12 +346,15 @@ pub fn set_random_binds(config: &mut Config) {
         .parse()
         .unwrap();
     let (rpc_port, p2p_port) = loop {
-        let mut rng = rand::thread_rng();
-        // Use a non-privileged port between 1024 and 65534
-        let rpc_port: u16 = rng.gen_range(1024..65533);
-        let p2p_port = rpc_port + 1;
-        if rpc_port != prior_rpc_port && p2p_port != prior_p2p_port {
-            break (rpc_port, p2p_port);
+        let a = rng.gen_range(1024..u16::MAX); // use a non-privileged port between 1024 and 65534
+        let b = rng.gen_range(1024..u16::MAX); // use a non-privileged port between 1024 and 65534
+        if a != b
+            && a != prior_rpc_port
+            && a != prior_p2p_port
+            && b != prior_rpc_port
+            && b != prior_p2p_port
+        {
+            break (a, b);
```

Comment:
Would it be worth refactoring this into a reusable function? Something like:

```rust
pub fn select_unused_port(used: &HashMap<u16, ()>) -> u16 {
    let mut rng = rand::thread_rng();
    loop {
        let port = rng.gen_range(1024..u16::MAX); // use a non-privileged port between 1024 and 65534
        if !used.contains_key(&port) {
            return port;
        }
    }
}
```

Then:
```rust
    let mut used_ports = HashSet::new();
    let rpc_port = select_unused_port(&used_ports);
    used_ports.insert(rpc_port);
    let p2p_port = select_unused_port(&used_ports);
```

Reply:
Killing me here Brice.

Reply:
Sorry!